### PR TITLE
Remove net462 TFM

### PR DESCRIPTION
**Pull request by bachuv**
Removed .NET Framework 4.6.2 TFM and upgraded DurableTask.Core version to 3.0.0.

### CODE REVIEW

File: src/DurableTask.AzureStorage/DurableTask.AzureStorage.csproj [hunk 1]
```diff
@@ -48,11 +48,6 @@
     <PackageReference Include="System.Linq.Async" Version="6.0.1" />
   </ItemGroup>
 
-  <ItemGroup Condition="'$(TargetFramework)' == 'net462'">
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="1.1.1" />
-    <PackageReference Include="WindowsAzure.Storage" version="7.2.1" />
-  </ItemGroup>
-
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="2.2.0" />
```
Comment (cgillum) [inline src/DurableTask.AzureStorage/DurableTask.AzureStorage.csproj:52]: The expression `Condition="'$(TargetFramework)' == 'netstandard2.0'"` is always going to be true now, so let's consolidate all package references to be in the same `<ItemGroup>` above.

File: src/DurableTask.Core/DurableTask.Core.csproj [hunk 2]
```diff
@@ -34,12 +34,6 @@
...
     <Version>$(VersionPrefix)-$(VersionSuffix)</Version>
   </PropertyGroup>
-  
-  <ItemGroup Condition="'$(TargetFramework)' == 'net462'">
-    <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="1.1.1" />
-    <!--Newtonsoft.Json 7.0.1 is kept for backwards compatibility with Functions runtime V1, so we supress warning NU1903. -->
-    <PackageReference Include="Newtonsoft.Json" Version="7.0.1" NoWarn="NU1903" />
-  </ItemGroup>
 
   <ItemGroup Condition="'$(TargetFramework)' == 'netstandard2.0'">
```
Comment (cgillum) [inline src/DurableTask.Core/DurableTask.Core.csproj:38]: Similar comment as in the other csproj file, let's consolidate all the `<PackageReference .../>` nodes to be in the same `<ItemGroup>`

**Review by davidmrdavid — CHANGES_REQUESTED** [review]
LGTM but let's make sure the GitHub actions are running on all PRs, as this has not had the validation executed on it :)

File: src/DurableTask.ServiceBus/Tracking/AzureTableOrchestrationHistoryEventEntity.cs [hunk 3]
```diff
@@ -35,11 +35,7 @@ internal class AzureTableOrchestrationHistoryEventEntity : AzureTableCompositeTa
         private static readonly JsonSerializerSettings ReadJsonSettings = new JsonSerializerSettings
         {
             TypeNameHandling = TypeNameHandling.Objects,
-#if NETSTANDARD2_0
```
Comment (nytian) [inline src/DurableTask.ServiceBus/Tracking/AzureTableOrchestrationHistoryEventEntity.cs:38]: Just wondering can we update the code in ServiceBus? Doesn't this belong to other teams?

**Review by nytian — APPROVED** [review]
Thanks for the PR! Left a small question :)

File: src/DurableTask.ServiceBus/Tracking/AzureTableOrchestrationHistoryEventEntity.cs (hunk 3 above)
Comment (davidmrdavid) [reply src/DurableTask.ServiceBus/Tracking/AzureTableOrchestrationHistoryEventEntity.cs:38]: ServiceBus is not actively maintained. It should be ok. I think you're thinking of ServiceFabric, which is maintained by another team.

**Review by davidmrdavid — APPROVED** [review]
I'll approve since the GitHub actions should run as soon as we try to merge to main. Let's open that PR asap

**Pull request merged**